curriculum/challenges/english/25-front-end-development/quiz-dynamic-programming/66edd5ffe490bef99ea99395.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --answer--
 
-Placeholder answer
+Dynamic programming is a method for solving complex problems by breaking them down into simpler subproblems and solving each subproblem just once, storing the solutions for future use.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What are the two main properties that a problem must have to be solved using dynamic programming?
 
 #### --distractors--
 
-Placeholder distractor 1
+The problem must have a fixed number of variables and a linear solution
 
 ---
 
-Placeholder distractor 2
+The problem must be solvable in polynomial time and have a unique solution
 
 ---
 
-Placeholder distractor 3
+The problem must involve sorting and searching algorithms
 
 #### --answer--
 
-Placeholder answer
+The problem must have optimal substructure and overlapping subproblems
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is memoization in dynamic programming?
 
 #### --distractors--
 
-Placeholder distractor 1
+Memoization is a technique for compressing data to save memory.
 
 ---
 
-Placeholder distractor 2
+Memoization is a method for sorting data in ascending order
 
 ---
 
-Placeholder distractor 3
+Memoization is a process of encrypting data for security purposes.
 
 #### --answer--
 
-Placeholder answer
+Memoization is a technique where you store the results of expensive function calls and reuse them when the same inputs occur again.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How does dynamic programming differ from divide and conquer?
 
 #### --distractors--
 
-Placeholder distractor 1
+Dynamic programming always uses a greedy approach to solve problems
 
 ---
 
-Placeholder distractor 2
+Divide and conquer algorithms do not use recursion.
 
 ---
 
-Placeholder distractor 3
+Dynamic programming is only used for sorting algorithms
 
 #### --answer--
 
-Placeholder answer
+Dynamic programming solves each subproblem only once and stores the results, while divide and conquer may solve the same subproblem multiple times.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the time complexity of the Fibonacci sequence using dynamic programming? 
 
 #### --distractors--
 
-Placeholder distractor 1
+O(n^2)
 
 ---
 
-Placeholder distractor 2
+O(log n)
 
 ---
 
-Placeholder distractor 3
+O(2^n)
 
 #### --answer--
 
-Placeholder answer
+O(n)
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is a greedy algorithm?
 
 #### --distractors--
 
-Placeholder distractor 1
+A greedy algorithm always finds the globally optimal solution.
 
 ---
 
-Placeholder distractor 2
+A greedy algorithm uses backtracking to find the best solution.
 
 ---
 
-Placeholder distractor 3
+A greedy algorithm is only used for sorting problems
 
 #### --answer--
 
-Placeholder answer
+A greedy algorithm is an approach for solving problems by making the locally optimal choice at each stage with the hope of finding a global optimum
```

Comment:
I think it would be better if this was a little less verbose. I mean it's correct. But I like the phrasing of "A greedy algorithm is an algorithm that finds a solution to problems in the shortest time possible"